### DISCUSSION TITLE
How to use custom Conda kernel in Code-Server

### DISCUSSION THREAD

**Discussion by RemiLacroix-IDRIS** [coder/code-server#6772] [Q&A]
Hi,

I am trying to use a custom Conda kernel in Code-Server (i.e. a Conda environment which is not installed in the default path) but I cannot figure out how to do that.

After installing the Python/Jupyter extensions I can see some Conda environments but not the ones I am interested in.

Is it possible to specify the path of specific Conda environment?

Best,
Rémi

**Comment by benz0li** [comment]
@RemiLacroix-IDRIS I might help if you create a minimal reproducible example at https://demo.jupyter.b-data.ch.  
ℹ️ At https://demo.jupyter.b-data.ch you can start images from my/b-data's [Jupyter](https://github.com/b-data/jupyterlab-julia-docker-stack)[Lab](https://github.com/b-data/jupyterlab-mojo-docker-stack) [docker](https://github.com/b-data/jupyterlab-python-docker-stack) [stacks](https://github.com/b-data/jupyterlab-r-docker-stack) which provide `code-server`.

**Comment by RemiLacroix-IDRIS** [comment]
Hello,

Sorry for the late answer, I'm not sure how to create an example as we use custom environments that we install locally in non-standard paths.

I haven't been able to start Code Server on the b-data server but I would guess that in that case Code Server is started from inside the environment while what we would like is using an external environment from Code Server.

Best,
Rémi

**Reply by benz0li** [reply]
> I haven't been able to start Code Server on the b-data server

Just click on the `code-server` icon.

> but I would guess that in that case Code Server is started from inside the environment while what we would like is using an external environment from Code Server.

Correct.

Inside the container you can create [virtual environments](https://docs.python.org/3/library/venv.html) or even [install Miniconda](https://docs.anaconda.com/miniconda/install/).  
👉 These environments can then be used in conjunction with the [Jupyter](https://open-vsx.org/extension/ms-toolsai/jupyter) extension.

**Comment by RemiLacroix-IDRIS** [comment]
> Just click on the code-server icon.

Sorry, I meant when I do that I either get a blank page or an error:
> The workbench failed to connect to the server (Error: WebSocket close with status code 1006)

> These environments can then be used in conjunction with the [Jupyter](https://open-vsx.org/extension/ms-toolsai/jupyter) extension.

We do have this extension and the "Python" extension installed but we haven't be able to figure out how to load an environment which is not detected by default.

**Reply by benz0li** [reply]
> The workbench failed to connect to the server (Error: WebSocket close with status code 1006)

Highly likely that you are behind a corporate firewall/proxy that does not allow websocket connections.

> We do have this extension and the "Python" extension installed but we haven't be able to figure out how to load an environment which is not detected by default.

IMHO *auto-detection* of Python/Conda environments only works if set up at *default* or *well known* locations.

**Reply by RemiLacroix-IDRIS** [reply]
> IMHO auto-detection of Python/Conda environments only works if set up at default or well known locations

Exactly, what we would like is being able to somehow load an env that isn't detected by default. I tried with the full path but that didn't seem to work.

**Reply by benz0li** [reply]
Please check https://github.com/microsoft/vscode-jupyter/discussions or https://github.com/microsoft/vscode-jupyter/issues.

This is related to the extension, not `code-server` itself.
